llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -447,7 +459,7 @@ def __init__(self, config: MPTConfig):
             raise ValueError(
                 'MPTForCausalLM only supports tied word embeddings')
 
-        self.transformer = MPTModel(config)
+        self.transformer: MPTModel = MPTModel(config)
```

Comment:
Why do this here?
Does this set precedent that ALL variables should be typed?
Or is this to help alleviate a bunch of typing issues?

Reply:
This alleviates a bunch of typing issues down the road -- since `transformer` is now an `MPTModel`, static type checkers know that accesses to its attributes are valid and can retrieve the typing, as opposed to resolving to `Unknown`